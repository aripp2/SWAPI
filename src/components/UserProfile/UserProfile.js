import React from 'react';
import './UserProfile.scss';

const UserProfile = () => {
  return (
      <header>User Profile</header>
    )
}

export default UserProfile;