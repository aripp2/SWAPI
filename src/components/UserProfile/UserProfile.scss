@import './src/variables.scss';

.profile {
  width: 20%;
  text-align: center;
  display: flex;
  flex-direction: column;
  font-family: $body-font;
  margin: 5px auto;
  
  h2 {
    font-size: 3em;
    font-family: $title-font;
    margin: 15px auto;
    letter-spacing: 7px;
  }

  h3 {
    font-size: 2em;
    font-weight: 700;
    margin: 10px auto;
  }

  h4 {
    font-size: 1.5em;
    font-weight: 700;
    margin: 10px auto;
  }

  h5 {
    font-size: 1.3em;
    margin: 5px auto;
  }

  button {
    height: 50px;
    width: 80%;
    font-size: 1.2em;
    font-family: $body-font;
    margin-top: 25px;
    background-color: #f0ffffba;
    border-radius: 25px;
    outline: none;
    &:hover {
      border: 5px solid $yellow-color;
      font-weight: 700;
    }
  }
}

