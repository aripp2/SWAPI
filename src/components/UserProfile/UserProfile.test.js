import React from 'react'; 
import { shallow } from 'enzyme'; 
import UserProfile from './UserProfile'

describe('UserProfile', () => {

  it('should match match snapshot', () => {
    const wrapper = shallow(
      <UserProfile favorites={[
        {
          films: [
            'The Empire Strikes Back',
            'Attack of the Clones',
            'The Phantom Menace',
            'Revenge of the Sith',
            'Return of the Jedi',
            'A New Hope'
          ],
          homeworld: 'Tatooine',
          name: 'C-3PO',
          population: '200000',
          species: 'Droid'
        },
        {
          films: [
            'The Empire Strikes Back',
            'Attack of the Clones',
            'The Phantom Menace',
            'Revenge of the Sith',
            'Return of the Jedi',
            'A New Hope',
            'The Force Awakens'
          ],
          homeworld: 'Naboo',
          name: 'R2-D2',
          population: '4500000000',
          pecies: 'Droid'
        }
      ]}/> )
      expect(wrapper).toMatchSnapshot(); 
  })

})