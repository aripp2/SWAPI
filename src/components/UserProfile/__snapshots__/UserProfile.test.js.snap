// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot 1`] = `
<aside
  className="profile"
>
  <h2>
    Star Wars Trivia
  </h2>
  <h3>
    Hi 
    Amy
    !
  </h3>
  <h4>
    Favorite Quote:
  </h4>
  <h5>
    "
    No clue
    "
  </h5>
  <h4>
    Knowledge Level:
  </h4>
  <h5>
    Star What?
  </h5>
  <Link
    to="/favorites"
  >
    <button>
      Favorites 
      0
    </button>
  </Link>
  <Link
    to="/"
  >
    <button>
      Sign Out
    </button>
  </Link>
  <Link
    to="/movies"
  >
    <button>
      All Movies
    </button>
  </Link>
</aside>
`;
